### ISSUE TITLE
Regression in nuitka 1.1+: loading precompiled module

### ISSUE THREAD

**Issue by Aksem** [Nuitka/Nuitka#1832]
**Problem:** compiled program crashes with SIGSEGV. The test program depends on [orjson](https://github.com/ijl/orjson) and just imports it([reference](https://github.com/Aksem/nuitka_load_module_regression/blob/main/nuitka_load_module_regression/main.py#L1)). So it fails to import the module.

orjson is implemented in Rust and is distributed as a pre-built library. It can be the reason because all libraries and modules compiled with nuitka work correctly. It's just my assumption, I'm not familiar with nuitka well enough yet.

Repository with minimal example to reproduce the issue: https://github.com/Aksem/nuitka_load_module_regression. It also includes configured CI to compile the program with nuitka 1.0.8 and 1.1. Version compiled with nuitka 1.0.8 works well, with 1.1 crashes.

**Nuitka version:** 1.1+ (tested 1.1, 1.1.1, 1.1.2, factory)
**Environment:** Python 3.10.7 installed via pyenv, nuitka installed via poetry. All OS: Linux/Ubuntu, macOS, Windows.
Nuitka version output:
```
1.1
Commercial: None
Python: 3.10.7 (main, Sep  6 2022, 15:19:58) [GCC 9.4.0]
Flavor: Unknown
Executable: /home/runner/.cache/pypoetry/virtualenvs/nuitka-load-module-regression-Ed2RT1Ut-py3.10/bin/python
OS: Linux
Arch: x86_64
Distribution: Ubuntu (based on Debian) 20.04.5
```

Nuitka parameters: `--standalone --follow-imports`([reference in example](https://github.com/Aksem/nuitka_load_module_regression/blob/main/.github/workflows/ci.yml#L94)).

Execution logs with nuitka 1.1+ and `--trace-execution` compilation parameter(last lines with orjson loading):
```
orjson: Calling createModuleConstants().
orjson: Calling createModuleCodeObjects().
Execute: /home/user/.cache/pypoetry/virtualenvs/app-mBho1MvJ-py3.10/lib64/python3.10/site-packages/orjson/__init__.py:1 <Node 'STATEMENT_ASSIGNMENT_VARIABLE_CONSTANT_IMMUTABLE' with {'variable_name': '__doc__', 'is_temp': False, 'var_type': 'object', 'owner': 'orjson'}>
Execute: /home/user/.cache/pypoetry/virtualenvs/app-mBho1MvJ-py3.10/lib64/python3.10/site-packages/orjson/__init__.py:1 <Node 'STATEMENT_ASSIGNMENT_VARIABLE_GENERIC' with {'variable_name': '__file__', 'is_temp': False, 'var_type': 'object', 'owner': 'orjson'}>
Execute: /home/user/.cache/pypoetry/virtualenvs/app-mBho1MvJ-py3.10/lib64/python3.10/site-packages/orjson/__init__.py:1 <Node 'STATEMENTS_FRAME_MODULE' with {'co_name': '<module orjson>', 'co_kind': 'Module', 'co_varnames': '', 'co_freevars': '', 'co_argcount': 0, 'co_posonlyargcount': 0, 'co_kwonlyargcount': 0, 'co_has_starlist': False, 'co_has_stardict': False, 'co_filename': 'orjson/__init__.py', 'co_lineno': 1, 'co_new_locals': False, 'co_is_optimized': False, 'code_flags': ''}>
Execute: /home/user/.cache/pypoetry/virtualenvs/app-mBho1MvJ-py3.10/lib64/python3.10/site-packages/orjson/__init__.py:1 <Node 'STATEMENT_ASSIGNMENT_VARIABLE_GENERIC' with {'variable_name': '__path__', 'is_temp': False, 'var_type': 'object', 'owner': 'orjson'}>
Execute: /home/user/.cache/pypoetry/virtualenvs/app-mBho1MvJ-py3.10/lib64/python3.10/site-packages/orjson/__init__.py:1 <Node 'STATEMENT_DICT_OPERATION_SET'>
Execute: /home/user/.cache/pypoetry/virtualenvs/app-mBho1MvJ-py3.10/lib64/python3.10/site-packages/orjson/__init__.py:1 <Node 'STATEMENT_ASSIGNMENT_ATTRIBUTE' with {'attribute_name': 'origin'}>
Execute: /home/user/.cache/pypoetry/virtualenvs/app-mBho1MvJ-py3.10/lib64/python3.10/site-packages/orjson/__init__.py:1 <Node 'STATEMENT_ASSIGNMENT_ATTRIBUTE' with {'attribute_name': 'has_location'}>
Execute: /home/user/.cache/pypoetry/virtualenvs/app-mBho1MvJ-py3.10/lib64/python3.10/site-packages/orjson/__init__.py:1 <Node 'STATEMENT_ASSIGNMENT_ATTRIBUTE' with {'attribute_name': 'submodule_search_locations'}>
Execute: /home/user/.cache/pypoetry/virtualenvs/app-mBho1MvJ-py3.10/lib64/python3.10/site-packages/orjson/__init__.py:1 <Node 'STATEMENT_ASSIGNMENT_VARIABLE_CONSTANT_IMMUTABLE' with {'variable_name': '__cached__', 'is_temp': False, 'var_type': 'object', 'owner': 'orjson'}>
Execute: /home/user/.cache/pypoetry/virtualenvs/app-mBho1MvJ-py3.10/lib64/python3.10/site-packages/orjson/__init__.py:1 <Node 'STATEMENT_IMPORT_STAR'>
fish: Job 1, './app.dist/app…' terminated by signal SIGSEGV (Address boundary error)
```

**Comment by kayhayen** [comment]
Thanks, I managed to reproduce it indeed, and am currently git bisecting it.

**Comment by kayhayen** [comment]
It seems, rust people feel error handling is optional, at least for Python bindings. Adding  `--include-module=zoneinfo` makes it work, as does this user yaml config

```
- module-name: "orjson"
  implicit-imports:
    - depends:
        - "zoneinfo"
```

I am going to make this part of the next hotfix. Not including standard library modules is a risk we are taking, where we probably need to make a list of them and abort standalone execution with a proper error. Standard library modules not included on purpose, can be deadly to C code only developed against CPython. We got an action going for this anyway, and since we want to only become more aggressive with this, maybe about time to do it.

**Comment by Aksem** [comment]
Thanks for the quick solution!

The factory branch doesn't work because the hotfix isn't merged, but branch hotfix/1.1.3 works, thanks again!

**Comment by kayhayen** [comment]
This is part of the hotfix release 1.1.3 that I just made.
